### PR TITLE
refactor: mark small functions as inline, and simplify unwrapping Result and Option

### PR DESCRIPTION
**Pull request by null8626**
Shortened code by:
- using `if let` statements for Option `match` statements that doesn't do anything for the `None` case
- propagating `Result` or `Option` using `?`

Also marked several small functions as inline to improve performance.

### CODE REVIEW

File: editor/src/components/tab_label/imp.rs [hunk 1]
```diff
@@ -18,10 +18,12 @@ impl ObjectSubclass for TabLabel {
...
 
+    #[inline]
     fn class_init(klass: &mut Self::Class) {
         Self::bind_template(klass);
     }
 
+    #[inline]
     fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
         obj.init_template();
     }
```
Comment (fortressia) [inline editor/src/components/tab_label/imp.rs:29]: Can you please remove `tab_label` completely? It's a dead code noww.

File: editor/src/components/window/file.rs [hunk 2]
```diff
@@ -182,22 +182,10 @@ impl FileImplementedEditor for super::EchidnaWindow {
...
-                        }
-                    }
-
-                    None => self.action_save_file_as(),
-                }
-            }
-            Err(e) => Err(e),
+        let page: EchidnaCoreEditor = self.get_current_tab()?;
+        match page.file() {
+            Some(_) => page.save_file(None),
```
Comment (fortressia) [inline editor/src/components/window/file.rs:187]: It seems I accidentally wrote `page.save_file(None)`. Can you change it to this?

```rs
page.save_file(file.location())
```

File: editor/src/components/sidebar/imp.rs [hunk 3]
```diff
@@ -16,10 +16,12 @@ impl ObjectSubclass for EchidnaSidebar {
     type Type = super::EchidnaSidebar;
     type ParentType = gtk::Box;
 
+    #[inline]
```
Comment (fortressia) [inline editor/src/components/sidebar/imp.rs:19]: I'm not sure whether we can inline a function in an implementation block. I will try to see whether Rustc will love it or not, do some QA things.

Perhaps I should also ask the folks at GTK Rs about this, but I don't knoww.

Comment (fortressia) [reply editor/src/components/sidebar/imp.rs:19]: @null8626 Does Rustc allow this? I think for basic type-checking, you don't need GTK installed in your system.

File: editor/src/components/tab_label/imp.rs (hunk 1 above)
Comment (null8626) [reply editor/src/components/tab_label/imp.rs]: Done https://github.com/null8626/Echidna/commit/c9122f2b8a39f73340328445788599dfc4c15be3

File: editor/src/components/window/file.rs (hunk 2 above)
Comment (null8626) [reply editor/src/components/window/file.rs]: Done https://github.com/null8626/Echidna/commit/1e492ba790917f4d7bc2de4feb5c456b306376e7

File: editor/src/components/sidebar/imp.rs (hunk 3 above)
Comment (null8626) [reply editor/src/components/sidebar/imp.rs:19]: Yes

Comment (null8626) [reply editor/src/components/sidebar/imp.rs:19]: i tested it first on a test file before creating this type of patches.

**Comment by fortressia** [comment]
LGTM <3

**Comment by null8626** [comment]
<3

File: editor/src/components/editor/mod.rs [hunk 4]
```diff
@@ -28,77 +28,70 @@ impl EchidnaCoreEditor {
...
-                }
-                Ok(this)
+                },
+                );
             }
-            Err(e) => Err(e),
         }
+
+        if let Some(a) = app {
+            let manager = a.unwrap().style_manager();
```
Comment (fortressia) [inline editor/src/components/editor/mod.rs:86]: We already have unwrapped `app` to `a`, so can you please remove the `.unwrap()`?

```
error[E0599]: no method named `unwrap` found for reference `&libadwaita::Application` in the current scope
  --> editor/src/components/editor/mod.rs:86:29
   |
86 |             let manager = a.unwrap().style_manager();
   |                             ^^^^^^ method not found in `&libadwaita::Application`
```

File: editor/src/components/window/file.rs [hunk 5]
```diff
@@ -182,22 +182,10 @@ impl FileImplementedEditor for super::EchidnaWindow {
     }
```
Comment (fortressia) [inline editor/src/components/window/file.rs:182]: Can you please do this as well?

```
warning: unused import: `fmt`
 --> editor/src/components/window/file.rs:7:25
  |
7 | use std::{error::Error, fmt};
  |                         ^^^
  |
  = note: `#[warn(unused_imports)]` on by default

```

Comment (null8626) [reply editor/src/components/window/file.rs:182]: done https://github.com/null8626/Echidna/commit/0e159c79bae10e868219cd3bff77355589bfb9ad

File: editor/src/components/editor/mod.rs (hunk 4 above)
Comment (null8626) [reply editor/src/components/editor/mod.rs]: done https://github.com/null8626/Echidna/commit/484ac7df1b68cef6308f2d6f81b4b25831a1f300

File: editor/src/components/window/file.rs [hunk 6]
```diff
@@ -182,22 +182,10 @@ impl FileImplementedEditor for super::EchidnaWindow {
...
-                        }
-                    }
-
-                    None => self.action_save_file_as(),
-                }
-            }
-            Err(e) => Err(e),
+        let page: EchidnaCoreEditor = self.get_current_tab()?;
+        match page.file() {
+            Some(_) => page.save_file(file.location()),
```
Comment (fortressia) [inline editor/src/components/window/file.rs:187]: I think we need to change the `Some(_)` to `Some(file)`
```
error[E0423]: expected value, found macro `file`
   --> editor/src/components/window/file.rs:187:39
    |
187 |             Some(_) => page.save_file(file.location()),
    |                                       ^^^^ not a value
```

Comment (null8626) [reply editor/src/components/window/file.rs]: done https://github.com/null8626/Echidna/commit/a094a21e4c8a3771221de02c994b773c5e6edc56

**Comment by fortressia** [comment]
Merged!

**Pull request closed without merging**